### ISSUE TITLE
Read line multibyte char

### ISSUE THREAD

**Issue by keepcosmos** [piotrmurach/tty-prompt#29]
I found some bug. When get multibyte input, it dese not work well.

I think problem is `Reader#read_line` get char as byte

For example
```ruby
"한글".bytes # Korean
# => [237, 140, 140, 235, 166, 172]

line = ''
[237, 140, 140, 235, 166, 172] .each { |b| line << b }
#=> "í\u0095\u009Cê¸\u0080"
```

Solution I think about,
```ruby
"한글".bytes.pack("C*").force_encoding("utf-8")

# or

line.codepoints.to_a.pack("C*").force_encoding("utf-8")
```

I suggested #28, but need more graceful way to fix it.
How should I work for this problem?

**Comment by piotrmurach** [comment]
Thank you for submitting this issue.

I would suggest as a first step to write test cases for all the Reader calls to show failures. We need to demonstrate any bugs before we get to fix them. Therefore, if you could write a single test for each of these [read_keypress_spec](https://github.com/piotrmurach/tty-prompt/blob/master/spec/unit/reader/read_keypress_spec.rb), [read_line](https://github.com/piotrmurach/tty-prompt/blob/master/spec/unit/reader/read_line_spec.rb) and [read_multiline_spec](https://github.com/piotrmurach/tty-prompt/blob/master/spec/unit/reader/read_multiline_spec.rb) that would be really helpful. Please submit PR just to show failures?

**Comment by piotrmurach** [comment]
Nicely done! As expected the problem affects all the reading calls. I would prefer smaller pull requests as they are easier to review and merge. Therefore, as a next step we could tackle only the `read_keypress` call, which means fixing the `read_char` call it relies on. Would you mind fixing this as a first step and submitting PR?

**Comment by keepcosmos** [comment]
@piotrmurach Ok, I will

**Comment by piotrmurach** [comment]
@keepcosmos Any progress?

**Comment by keepcosmos** [comment]
I have no idea how to change `while CSI.start_with?(chars) || chars.start_with?(CSI) && !(64..126).include?(chars.each_codepoint.to_a.last)` in `#read_char`. any idea?

**Comment by piotrmurach** [comment]
Atm I'm recovering from illness but once feeling better I will look into this.

**Comment by keepcosmos** [comment]
@piotrmurach Thanks. Take care yourself. 👍

**Comment by piotrmurach** [comment]
Released `v0.9.0` with the fix. I'm super excited by your jira cli tool and thanks for using tty-prompt!

**Comment by keepcosmos** [comment]
@piotrmurach Thanks! tty series are super super helpful for making Terjira :)
